### PR TITLE
[FIX] mrp: correctly calculate producible_qty Ready in MO Overview

### PR DESCRIPTION
**Pull request by ticodoo**
Previous to this commit, the qty reserved was only considering the currently reserved quantities in the MO, not the ones calculated and displayed in the MO Overview (i.e. including the quantities reserved in the picking during 2 step mrp). This would lead to showing reserved quantities but a "Not Ready" status, which is confusing and inconsistent.

Now we pass the calculated component quantities to the method that calculates this state and ensure the calculation is consistent.

Steps to reproduce:
- enable 2 or 3 step mrp
- create an MO with a component that is in stock at the stock loc
- confirm MO and open "Overview"

Also change "Quantity" back to "Reserved" in the pdf version of the MO Overview. This was a mistake change since this column name already exists and "Reserved" will be hopefully less confusing for users.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/report/mrp_report_mo_overview.py [hunk 1]
```diff
@@ -543,6 +542,7 @@ def _get_replenishment_lines(self, production, move_raw, replenish_data, level,
             else:
                 replenishment['summary']['receipt'] = self._check_planned_start(production.date_start, self._get_replenishment_receipt(doc_in, replenishment.get('components', [])))
             replenishment['summary']['mo_cost_decorator'] = self._get_comparison_decorator(replenishment['summary']['real_cost'], replenishment['summary']['mo_cost'], replenishment['summary']['currency'].rounding)
+            replenishment['summary']['formatted_state'] = doc_in._name == 'mrp.production' and self._format_state(doc_in, replenishment['components']) or self._format_state(doc_in)
```
Comment (clesgow) [inline addons/mrp/report/mrp_report_mo_overview.py:545]:
```suggestion
            replenishment['summary']['formatted_state'] = self._format_state(doc_in, replenishment['components']) if doc_in._name == 'mrp.production' else self._format_state(doc_in)
```
I find it slightly more readable, but it's up to you.

File: addons/mrp/report/mrp_report_mo_overview.py [hunk 2]
```diff
@@ -191,22 +191,22 @@ def _get_mo_summary(self, production, components, current_mo_cost, current_real_
...
-        for product_id, comp_qty_to_produce in components_qty_to_produce.items():
-            if float_is_zero(comp_qty_to_produce, precision_rounding=product_id.uom_id.rounding):
+        for component in components:
+            component = component["summary"]
+            # skip consumables + products with nothing to consume
+            if isinstance(component["quantity_free"], bool) or float_is_zero(component["quantity"], precision_rounding=component["uom_precision"]):
                 continue
             comp_producible_qty = float_round(
-                record.product_qty * (components_qty_reserved[product_id] + product_id.free_qty) / comp_qty_to_produce,
+                record.product_qty * (component["quantity_reserved"] + component["quantity_free"]) / component["quantity"],
```
Comment (clesgow) [inline addons/mrp/report/mrp_report_mo_overview.py:209]: The previous loop (to know how many components are needed in total by product) made sense, if you have a disparity between quantities required for the same component.

e.g. the following case :
image
The first move will get the full 10 available in its reservation, but the second one will have 0, meaning the second iteration will return 'Not Ready'.
But it's incorrect, as there are enough components for 1 finished product.

By changing that, it should also avoid counting the `free_qty` twice if there is some quantity available for a duplicated component. :thinking:

File: addons/mrp/report/mrp_report_mo_overview.py (hunk 1 above)
Comment (ticodoo) [reply addons/mrp/report/mrp_report_mo_overview.py]: you're the one who'll probably be reading it the most :wink: (I changed it)

File: addons/mrp/report/mrp_report_mo_overview.py (hunk 2 above)
Comment (ticodoo) [reply addons/mrp/report/mrp_report_mo_overview.py]: D'oh, I knew there was a reason why I did it that way in the first place

**Comment by clesgow** [comment]
@robodoo r+

**Comment by clesgow** [comment]
@robodoo rebase-ff

**Pull request closed without merging**